src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -253,18 +284,27 @@ def process_delayed_workflows(
     dcg_to_groups, trigger_type_to_dcg_model = get_dcg_group_workflow_detector_data(
         workflow_event_dcg_data
     )
-    dcg_to_workflow = trigger_type_to_dcg_model[DataConditionHandlerType.WORKFLOW_TRIGGER]
+    dcg_to_workflow = trigger_type_to_dcg_model[DataConditionHandlerType.WORKFLOW_TRIGGER].copy()
     dcg_to_workflow.update(trigger_type_to_dcg_model[DataConditionHandlerType.ACTION_FILTER])
 
     _, workflows_to_envs = fetch_workflows_envs(list(dcg_to_workflow.values()))
     data_condition_groups = fetch_data_condition_groups(list(dcg_to_groups.keys()))
 
-    _ = get_condition_query_groups(
+    # Get unique query groups to query Snuba
+    condition_groups = get_condition_query_groups(
```

Comment:
i think i removed this on accident when pushing the previous PR, there are tests in `TestGetSnubaResults`